common/src/main/java/org/figuramc/figura/mixin/render/renderers/SkullBlockRendererMixin.java

```diff
@@ -89,11 +88,25 @@ public void render(SkullBlockEntity skullBlockEntity, float f, PoseStack poseSta
 
     @Override
     public boolean shouldRenderOffScreen(SkullBlockEntity blockEntity) {
-        return avatar == null ? BlockEntityRenderer.super.shouldRenderOffScreen(blockEntity) : avatar.permissions.get(Permissions.OFFSCREEN_RENDERING) == 1;
+    	// if we have an avatar, check permissions
+    	if(avatar != null  && avatar.permissions != null) {
+    		// if we have permissions to render, then we render, else we don't render.
+    		if(avatar.permissions.get(Permissions.OFFSCREEN_RENDERING) == 1) {
+    		    return true;
+    		}
+    		return false;
+    	}
+    	//no avatar present, default rendering used.  
+    	return BlockEntityRenderer.super.shouldRenderOffScreen(blockEntity);
     }
 
     @Inject(at = @At("HEAD"), method = "getRenderType")
     private static void getRenderType(SkullBlock.Type type, GameProfile profile, CallbackInfoReturnable<RenderType> cir) {
-        avatar = profile != null && profile.getId() != null ? AvatarManager.getAvatarForPlayer(profile.getId()) : null;
+    	// reset avatar for skull
+    	avatar = null;
```

Comment:
This whole thing can be simplified to
```suggestion
    	avatar = profile != null && profile.getId() != null ? AvatarManager.getAvatarForPlayer(profile.getId()) : null;
```
There is literally no reason to expand the logic, the ternary is simpler and more compact, if you don't understand these you might want to review on some of the Java basics again because they are a useful tool and a basic operation. This change is unnecessary, the behavior is the exact same.

Reply:
I was always taught that if you are doing more than one if condition in a ternary you should expand the logic or use braces  to make the code easier for other developers to  read. I'll change it to a ternary with braces if  that's ok?